pkg/labels/selector_test.go

```diff
@@ -306,65 +493,89 @@ func TestRequirementLabelSelectorMatching(t *testing.T) {
 		}}, true, true},
 		{"x in (abc,def)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet("abc", "def"), t),
+			getRequirement("x", In, util.NewStringSet("abc", "def"), SetBasedMatch, t),
 		}}, true, true},
 		{"x in (abc,)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet("abc", ""), t),
+			getRequirement("x", In, util.NewStringSet("abc", ""), SetBasedMatch, t),
 		}}, true, true},
 		{"x in ()", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet(""), t),
+			getRequirement("x", In, util.NewStringSet(""), SetBasedMatch, t),
 		}}, true, true},
 		{"x not in (abc,,def),bar,z in (),w", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", NotIn, util.NewStringSet("abc", "", "def"), t),
-			getRequirement("bar", Exists, nil, t),
-			getRequirement("z", In, util.NewStringSet(""), t),
-			getRequirement("w", Exists, nil, t),
+			getRequirement("bar", Exists, nil, SetBasedMatch, t),
+			getRequirement("w", Exists, nil, SetBasedMatch, t),
+			getRequirement("x", NotIn, util.NewStringSet("abc", "", "def"), SetBasedMatch, t),
+			getRequirement("z", In, util.NewStringSet(""), SetBasedMatch, t),
 		}}, true, true},
 		{"x,y in (a)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("y", In, util.NewStringSet("a"), t),
-			getRequirement("x", Exists, nil, t),
+			getRequirement("y", In, util.NewStringSet("a"), SetBasedMatch, t),
+			getRequirement("x", Exists, nil, SetBasedMatch, t),
 		}}, false, true},
+		{"x=a", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", In, util.NewStringSet("a"), ExactMatch, t),
+		}}, true, true},
+		{"x=a,y!=b", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", In, util.NewStringSet("a"), ExactMatch, t),
+			getRequirement("y", NotIn, util.NewStringSet("b"), ExactMatch, t),
+		}}, true, true},
+
+		{"x=a||y=b", &LabelSelector{Requirements: []Requirement{}}, false, false},
 		{"x,,y", nil, true, false},
 		{",x,y", nil, true, false},
 		{"x nott in (y)", nil, true, false},
-		{"x not in ( )", nil, true, false},
-		{"x not in (, a)", nil, true, false},
+		{"x not in ( )", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", NotIn, util.NewStringSet(""), SetBasedMatch, t),
+		}}, true, true},
+		{"x not in (, a)", &LabelSelector{Requirements: []Requirement{
+
+			getRequirement("x", NotIn, util.NewStringSet("", "a"), SetBasedMatch, t),
+		}}, true, true},
 		{"a in (xyz),", nil, true, false},
 		{"a in (xyz)b not in ()", nil, true, false},
-		{"a ", nil, true, false},
-		{"a not in(", nil, true, false},
+		{"a ", &LabelSelector{Requirements: []Requirement{
+			getRequirement("a", Exists, nil, SetBasedMatch, t),
+		}}, true, true},
+		{"a in (xyz abc)", nil, false, false},  // no comma
+		{"a in (x,y,z,in)", nil, false, false}, // operator 'in' inside list of identifiers
```

Comment:
in the actual PR only `in` and `not` are keywords (in the usual sense). It's true that we can add a `context` info to remove this limitation as well. This will permit to specify something like:
`answer not in (not)` which looks funny but it makes sense.
OK.